internal/pkg/agent/protection/action.go

```diff
@@ -21,7 +21,7 @@ var (
 )
 
 type fleetActionWithAgents struct {
-	ActionID         string          `json:"action_id"` // Note the action_id here, since the signed action uses action_id for id
```

Comment:
@aleksmaus is it the ES schema or the API schema? The API schema is `id` and now the whole agent also used `id` instead of `action_id`.

Reply:
As far as I remember, it is ES field name, and I think the signed JSON payload uses action_id as well.

Reply:
So if I understood correctly, for those actions,there is an action inside the action. Is that correct?
[Here](https://github.com/elastic/elastic-agent/pull/4441/files#diff-401f5547d841bfdeff1bf4f32bba334bea2bff7d1147525983e726d601bfe785L64-L65) you're deserialising a whole action from the `data` field.
Is it just a copy of the action or something else?

Another question. Is there any reason for copying the action model to this package instead of using the models from `fleetapi`. Part of this PR consists on unifying action serialisation/deserialisation on the `fleetapi` package.

Reply:
Here is some old example I found of the signed action:
```
{
    "@timestamp": "2023-06-12T21:02:08.783Z",
    "expiration": "2023-07-12T21:02:08.783Z",
    "agents": [
        "f19a072f-49c9-4a1d-94f7-9c1c519eab19"
    ],
    "action_id": "164a6819-5c58-40f7-a33c-821c98ab0a8c",
    "data": {
        "version": "8.9.0"
    },
    "type": "UPGRADE",
    "traceparent": "00-baf54d4e5b51487c5c7dd520eef3ba6f-6959cd1c34a1bfd9-01",
    "signed": {
        "data": "eyJAdGltZXN0YW1wIjoiMjAyMy0wNi0xMlQyMTowMjowOC43ODNaIiwiZXhwaXJhdGlvbiI6IjIwMjMtMDctMTJUMjE6MDI6MDguNzgzWiIsImFnZW50cyI6WyJmMTlhMDcyZi00OWM5LTRhMWQtOTRmNy05YzFjNTE5ZWFiMTkiXSwiYWN0aW9uX2lkIjoiMTY0YTY4MTktNWM1OC00MGY3LWEzM2MtODIxYzk4YWIwYThjIiwiZGF0YSI6eyJ2ZXJzaW9uIjoiOC45LjAifSwidHlwZSI6IlVQR1JBREUiLCJ0cmFjZXBhcmVudCI6IjAwLWJhZjU0ZDRlNWI1MTQ4N2M1YzdkZDUyMGVlZjNiYTZmLTY5NTljZDFjMzRhMWJmZDktMDEifQ==",
        "signature": "MEQCIGxsrI742xKL6OSIfT6abkBfH+plFiTSO26RHNV1pHE8AiBxXgt3LlIowCiLZ/ayFqQWk0ge+c/J1L58csqg6R9nyA=="
    }
} 
```

The ```data``` field contains the base64 encoded action JSON that is signed by Kibana. This is the source of truth for the signed actions for the tamper protection for Endpoint for example. The agent initially had signature validation, but it was commented out for the initial release due to risks concerns raised by PM. So the Agent just passes the action through to the beats/Endpoint leaving it up to them to parse and validate the signed payload if it is present.

Reply:
Yes it is Kibana only by design for end-to-end signing. Only Kibana has the key(s) to sign the policy and actions. The goal was to prevent any tampering in the middle.

Reply:
Would need some good end to end test coverage with Endpoint here in order to make sure the schema changes will not break Endpoint.

Reply:
> Would need some good end to end test coverage with Endpoint here in order to make sure the schema changes will not break Endpoint.

You mean future schema changes or the ones in this PR?

The changes here should not have affected the actions schema outside the state/action store. The changes are to match fleet-server schema.

Reply:
I tested, the signed actions aren't affected. They still working

Reply:
Thank you!